src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/LambdaInvokeRequestHandler.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Concurrent;
+using System;
+using System.Threading.Tasks;
+using NewRelic.Agent.Api;
+using NewRelic.Agent.Api.Experimental;
+using NewRelic.Agent.Extensions.Providers.Wrapper;
+using NewRelic.Reflection;
+using NewRelic.Agent.Extensions.Helpers;
+
+namespace NewRelic.Providers.Wrapper.AwsSdk
+{
+    internal static class LambdaInvokeRequestHandler
+    {
+        private static ConcurrentDictionary<Type, Func<object, object>> _getResultFromGenericTask = new();
+        private static ConcurrentDictionary<string, string> _arnCache = new ConcurrentDictionary<string, string>();
+
+        private static object GetTaskResult(object task)
+        {
+            if (((Task)task).IsFaulted)
+            {
+                return null;
+            }
+
+            var getResponse = _getResultFromGenericTask.GetOrAdd(task.GetType(), t => VisibilityBypasser.Instance.GeneratePropertyAccessor<object>(t, "Result"));
+            return getResponse(task);
+        }
+
+        private static void SetRequestIdIfAvailable(IAgent agent, ITransaction transaction, dynamic response)
+        {
+            try
+            {
+                dynamic metadata = response.ResponseMetadata;
+                string requestId = metadata.RequestId;
+                transaction.AddCloudSdkAttribute("aws.requestId", requestId);
+            }
+            catch (Exception e)
+            {
+                agent.Logger.Debug(e, "Unable to get RequestId from response metadata.");
+            }
+        }
+
+        public static AfterWrappedMethodDelegate HandleInvokeRequest(InstrumentedMethodCall instrumentedMethodCall, IAgent agent, ITransaction transaction, dynamic request, bool isAsync, string region)
+        {
+            string functionName = request.FunctionName;
+            string qualifier = request.Qualifier;
+            if (!string.IsNullOrEmpty(qualifier) && !functionName.EndsWith(qualifier))
+            {
+                functionName = $"{functionName}:{qualifier}";
+            }
+            string arn;
+            if (!_arnCache.TryGetValue(functionName, out arn))
+            {
+                arn = AwsSdkHelpers.ConstructArn(agent, functionName, region, "");
+                _arnCache.TryAdd(functionName, arn);
+            }
+            var segment = transaction.StartTransactionSegment(instrumentedMethodCall.MethodCall, "InvokeRequest");
+            segment.GetExperimentalApi().MakeLeaf();
```

Comment:
This will suppress the HttpClient instrumentation, so we will no longer get the distributed tracing headers added, or any of the other external call attributes that were previously collected for these calls (prior to this instrumentation).

Reply:
That's correct, and the expected behavior. If we leave in the HttpClient segment, an additional Entity is created that can't be linked to the Lambda itself, because the URI is not unique enough. I take your point about the missing attributes, though, and will double check with the other devs on this initiative..

Reply:
Yes, and we need to ensure that we are still generating the expected metrics so that the externals UI will work, and the span.kind is correct for the span that is ultimately generated.